src/components/modals/SavedViewsModal.svelte

```diff
@@ -0,0 +1,142 @@
+    { field: 'owner', filter: 'text', headerName: 'Owner', resizable: true, sortable: true },
+    { field: 'updated_at', filter: 'text', headerName: 'Last Edited', resizable: true, sortable: true },
+    {
+      cellClass: 'action-cell-container',
+      cellRenderer: (params: ViewCellRendererParams) => {
+        const isEditable = params.data.owner !== 'system';
+        const actionsDiv = document.createElement('div');
+        actionsDiv.className = 'actions-cell';
+        new DataGridActions({
+          props: {
+            ...(isEditable
+              ? {
+                  deleteCallback: params.deleteView,
+                  deleteTooltip: {
+                    content: 'Delete View',
+                    placement: 'bottom',
+                  },
+                }
+              : {}),
+            rowData: params.data,
+            viewCallback: params.openView,
+            viewTooltip: {
+              content: 'Open View',
+              placement: 'bottom',
+            },
+          },
+          target: actionsDiv,
+        });
+
+        return actionsDiv;
+      },
+      cellRendererParams: {
+        deleteView,
+        openView,
+      } as CellRendererParams,
+      field: 'actions',
+      headerName: '',
+      resizable: false,
+      sortable: false,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 55,
+    },
+  ];
+
+  async function deleteView({ id: viewId }: View) {
+    const success = await effects.deleteView(viewId);
+
+    if (success) {
+      if ($view.id === viewId) {
+        goToNextView();
+      }
+    }
+  }
+
+  async function deleteViews({ detail: views }: CustomEvent<View[]>) {
+    const ids = views.map(({ id }) => id);
+    const success = await effects.deleteViews(ids);
+
+    if (success) {
+      if (ids.includes($view.id)) {
+        goToNextView();
+      }
+    }
+  }
+
+  async function goToNextView() {
+    const nextView = await effects.getView(null);
+    initializeView({ ...nextView });
+    setQueryParam('viewId', `${nextView.id}`);
+  }
+
+  async function openView({ id: viewId }: View) {
+    const query = new URLSearchParams(`?viewId=${viewId}`);
+    const newView = await effects.getView(query);
+
+    if (view) {
+      initializeView({ ...newView });
+      setQueryParam('viewId', `${newView.id}`);
+      dispatch('close');
+    } else {
+      console.log(`No view found for ID: ${viewId}`);
+    }
+  }
+</script>
+
+<Modal {height} {width}>
+  <ModalHeader on:close>Saved Views</ModalHeader>
+  <ModalContent>
+    <BulkActionDataGrid
+      {columnDefs}
+      isRowSelectable={rowData => rowData.data.owner !== 'system'}
+      items={$views}
+      pluralItemDisplayText="Views"
+      singleItemDisplayText="View"
+      on:bulkDeleteItems={deleteViews}
+      on:rowDoubleClicked={event => openView(event.detail)}
+    />
+  </ModalContent>
+  <ModalFooter>
+    <button class="st-button" on:click={() => dispatch('close')}> Close </button>
```

Comment:
Do we need a close button when there is no primary action? We don't use one for the merge requests modal, just the header X button. Could include it here but maybe we'd want to update it in merge requests modal to be consistent then? Though not sure we need it.

Reply:
Ah. I was trying to keep consistent with the About/ExpansionSequence modals, which do have the just close button. I didn't catch that the merge requests modal didn't have it. I could remove if it you feel strongly enough about it @AaronPlave.